homeassistant/components/home_connect/select.py

```diff
@@ -0,0 +1,286 @@
+        "LaundryCare.Washer.Program.DrumClean",
+        "LaundryCare.WasherDryer.Program.Cotton",
+        "LaundryCare.WasherDryer.Program.Cotton.Eco4060",
+        "LaundryCare.WasherDryer.Program.Mix",
+        "LaundryCare.WasherDryer.Program.EasyCare",
+        "LaundryCare.WasherDryer.Program.WashAndDry60",
+        "LaundryCare.WasherDryer.Program.WashAndDry90",
+    )
+}
+
+PROGRAMS_TRANSLATION_KEYS_MAP = {
+    value: key for key, value in TRANSLATION_KEYS_PROGRAMS_MAP.items()
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Home Connect select entities."""
+
+    def get_entities() -> list[HomeConnectProgramSelectEntity]:
+        """Get a list of entities."""
+        entities: list[HomeConnectProgramSelectEntity] = []
+        hc_api: ConfigEntryAuth = hass.data[DOMAIN][config_entry.entry_id]
+        programs_not_found = set()
+        for device in hc_api.devices:
+            if device.appliance.type in APPLIANCES_WITH_PROGRAMS:
+                with contextlib.suppress(HomeConnectError):
+                    programs = device.appliance.get_programs_available()
+                    if programs:
+                        for program in programs:
+                            if program not in PROGRAMS_TRANSLATION_KEYS_MAP:
+                                if program not in programs_not_found:
+                                    _LOGGER.warning(
+                                        'The program "%s" is not part of the official Home Connect API specification',
+                                        program,
+                                    )
+                                    programs_not_found.add(program)
+                        entities.extend(
+                            HomeConnectProgramSelectEntity(
+                                device, programs, start_on_select
+                            )
+                            for start_on_select in (True, False)
+                        )
+        return entities
+
+    async_add_entities(await hass.async_add_executor_job(get_entities), True)
+
+
+class HomeConnectProgramSelectEntity(HomeConnectEntity, SelectEntity):
+    """Select class for Home Connect programs."""
+
+    def __init__(
+        self, device: HomeConnectDevice, programs: list[str], start_on_select: bool
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(
+            device,
+            SelectEntityDescription(
+                key=BSH_ACTIVE_PROGRAM if start_on_select else BSH_SELECTED_PROGRAM,
+                translation_key="active_program"
+                if start_on_select
+                else "selected_program",
+            ),
+        )
+        self._attr_options = [
+            program_translation_key
+            for program in programs
+            if (program_translation_key := PROGRAMS_TRANSLATION_KEYS_MAP.get(program))
```

Comment:
We can pop the program from the `programs` list already when checking the membership above before creating the entities.